src/noisepy/monitoring/attenuation_utils.py

```diff
@@ -136,20 +137,202 @@ def check_s0(x: float) -> None:
+        Eobs_temp: The observed energy density function
+    ----------------------------------------------
+    """
+
+    monito_config = ConfigParameters_monitoring()
+    ONESTATION = monito_config.single_station
+
+    # initialize the temporary arrays
+    Esyn_temp = np.zeros((npts // 2 + 1))
+    Eobs_temp = np.zeros((npts // 2 + 1))
+
+    Eobs_temp = fmsv_single
+    # calculate the Esyn and SSR for combination of mean_free_path and intrinsic_b
+    for twn in range(npts // 2):
+        tm = dt * twn
+        s0 = cvel**2 * tm**2 - r**2
+        if s0 <= 0:
+            # logger.warning(f"s0 {s0} <0")
+            continue
+
+        if ONESTATION:
+            tmp = ESYN_RadiaTrans_onesta(mean_free_path, tm, r, cvel)
+        else:
+            tmp = ESYN_RadiaTrans_intersta(mean_free_path, tm, r, cvel)
+
+        Esyn_temp[twn] = tmp * math.exp(-1 * intrinsic_b * tm)
+
+    return Esyn_temp, Eobs_temp
+
+
+# ---
+def scaling_Esyn(Esyn_temp: np.ndarray, Eobs_temp: np.ndarray, twindow: np.ndarray, dt: float) -> np.ndarray:
+    """
+    # Scaling the Esyn in the specific window
+    ----------------------------------------------
+    Parameters:
+        Esyn_temp: The synthetic energy density function
+        Eobs_temp: The observed energy density function
+        twindow:   The window begin and end time array
+        dt:        The sampling interval
+    ----------------------------------------------
+    Return:
+        Esyn_temp: The scaled synthetic energy density function
+    ----------------------------------------------
+    """
+
+    # find the scaling factor in the specific window
+    SSR_temppp = np.zeros((len(twindow)))
+    for tsn in range(len(twindow)):
+        tsb = int(twindow[tsn] // dt)
+        SSR_temppp[tsn] = math.log10(Eobs_temp[tsb]) - math.log10(Esyn_temp[tsb])
+    crap = np.mean(SSR_temppp)
+    Esyn_temp *= 10**crap  # scale the Esyn
+    # using scalar factor for further fitting processes --> shape matters more than amplitude
+
+    return Esyn_temp, 10**crap
+
+
+# ### -----
+def windowing_SSR(
+    coda_window: np.array,
+    mean_free_path: float,
+    intrinsic_b: float,
+    fmsv: np.ndarray,
+    dist: np.ndarray,
+    npts: int,
+    dt: float,
+    cvel: float,
+    ncoda: int,
+) -> float:
+    """
+    # Calculate the sum of squared residuals (SSR) in the specific window
+    ----------------------------------------------
+    Parameters:
+        Eobs: The observed energy density function
+        Esyn: The synthetic energy density function
+        coda_window: The window begin and end time array
+        mean_free_path: The mean free path
+        intrinsic_b: The intrinsic absorption parameter b
+        npts: The total sample number
+    ----------------------------------------------
+    Return:
+        window_SSR: The sum of squared residuals in the specific window
+    """
+    Esyn_temp = np.zeros((npts // 2))
+    Eobs_temp = np.zeros((npts // 2))
+    PLOT_CHECK = False
```

Comment:
Considering moving this as the function arguments.

Reply:
Thanks for the suggestion. I will add the `PLOT_CHECK` as a function argument. `Esyn_temp` and `Eobs_temp` will keep recalculating via get_energy_density based on the parameters. Maybe we can leave them as they are.